packages/core/src/core.ts

```diff
@@ -793,7 +793,9 @@ class ThresholdKey implements ITKey {
 
     // update metadataStore
     this.metadata.setTkeyStoreDomain(moduleName, rawTkeyStoreItems);
-    await this.syncShareMetadata();
+
+    const shouldSync = sync === null || sync === undefined ? true : sync;
```

Comment:
checkout addShareDescription. implement in the same way.

Reply:
essentially modify all setTkeyStore to (_, true)

Reply:
Any reason for this? Because solution-wise, they're similar, but what you're proposing will introduce breaking changes, while current implementation keeps things work the way it has already been.

Reply:
how would that be a breaking change?

Reply:
I would like to maintain symmetry, hence the suggestion.

Reply:
Before:
- All `setTKeyStoreItem(name, item)` will update metadata.

After:
- `setTKeyStoreItem(name, item)` won't update metadata.

This breaks all existing implementation with assumption that `setTKeyStoreItem(name, item)` will update metadata.

Reply:
Updated.